lib/context_adapter_helper.js

```diff
@@ -333,12 +408,13 @@ function getOperationDescriptor(request, callback) {
 /**
  * Checks that an queryContext response returned by the Context Adapter to get
  *  service information is well-formed
+ * @param {object} context The context for logging
  * @param {object} response The response received from the Context Adapter
  * @param {string} serviceId The identifier of the service whose information was retrieved
  * @param {function} callback Callback to be called with the result of the check
  * @return {*} True if the queryContextResponse is valid. False otherwise.
  */
-function checkQueryContextResponse(response, serviceId, callback) {
+function checkQueryContextResponse(context, response, serviceId, callback) {
```

Comment:
Considering you are passing the response object already, it makes sense to do that. We use domains for the same process (but domains seem to have fallen in disgrace, so maybe the response object is a better place to set it). I consider the Correlator and the Context to be slightly different things, by the way (the correlator is for every request, and the context is the local, i.e.: module context).